scripts/metadata_comparison/test/test_digester.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python3
+
+import os
+import unittest
+from typing import AnyStr, Callable
+from metadata_comparison.lib.comparison_paths import ComparisonPath
+from pathlib import Path
+import logging
+from metadata_comparison.digester import digest
+from metadata_comparison.lib.logging import quieten_chatty_imports, set_log_verbosity
+
+
+def read_resource(filename: AnyStr) -> AnyStr:
+    path = Path('test/resources') / filename
+    with open(path, 'r') as file:
+        data = file.read()
+    return data
+
+
+class DigesterTestMethods(unittest.TestCase):
+    set_log_verbosity(verbose=True)
+    quieten_chatty_imports()
+
+    def test_digestion(self) -> None:
+        subdir = 'exome_germline_single_sample_v1.3/PAPIv2_alpha1/v1_style_machine_types'
+        local_parent = ComparisonPath.create(subdir)
+        gcs_parent = ComparisonPath.create(f'gs://papi-performance-analysis/{subdir}')
+        samples = {
+            'dev_C1963.CHMI_CHMI3_Nex1': {
+                'total_jobs': 133,
+                'more_than_1_attempts': 12,
+                'more_than_2_attempts': 1,
+                'more_than_3_attempts': 0,
+                'cromwell_time_more_than_3_minutes_longer_total': 21,
+                'cromwell_time_more_than_4_minutes_longer_total': 7,
+                'cromwell_time_more_than_5_minutes_longer_total': 4,
+                'cromwell_time_more_than_6_minutes_longer_total': 2,
+                'cromwell_time_more_than_7_minutes_longer_total': 1,
+                'cromwell_time_more_than_8_minutes_longer_total': 0,
+                # insert more intelligent assertions here
+            }
+            # more samples if needed
+            # 'dev_C862.NA19238',
+            # 'dev_D5327.NA12878',
+            # 'dev_D5327.NA12891',
+            # 'dev_D5327.NA12892',
+            # 'dev_RP-1535.NA17-308'
+        }
+
+        for sample_name in samples.keys():
+            # Copy down workflow and PAPI operations metadata from GCS if needed to test Local
+            local_sample_path = local_parent / sample_name
+            if not local_sample_path.exists():
+                logging.info(f"Local sample directory '{local_sample_path}' does not exist, downloading from GCS.")
```

Comment:
Is this expected/likely? Seems like a lot of logic to work around us accidentally deleting files from the repo?

Reply:
The files are not checked into the repo, they're only stored in GCS (this is the "real" metadata from the spike). The metadata files are downloaded if not already present to run the local version of the test. The local version is much nicer to iterate on than GCS since it's so much faster.

All of this would have been a lot more clear with some docs...

Reply:
Would it make sense to use the existing raw metadata/PAPI logs in the `resources` directory for test cases?

Reply:
The indirection was intentional since I wasn't sure that data belonged in a public repo.

Reply:
And also it should be staged in GCS anyway for the "real" GCS-based testing. 🙂